builtin/providers/azurerm/resource_arm_app_gateway.go

```diff
@@ -0,0 +1,1667 @@
+	d.Set("disabled_ssl_protocols", flattenAppGatewaySslPolicy(appGateway.ApplicationGatewayPropertiesFormat.SslPolicy))
+	d.Set("gateway_ip_configuration", flattenAppGatewayIPConfigurations(appGateway.ApplicationGatewayPropertiesFormat.GatewayIPConfigurations))
+	d.Set("frontend_port", flattenAppGatewayFrontendPorts(appGateway.ApplicationGatewayPropertiesFormat.FrontendPorts))
+	d.Set("frontend_ip_configuration", flattenAppGatewayFrontendIPConfigurations(appGateway.ApplicationGatewayPropertiesFormat.FrontendIPConfigurations))
+	d.Set("backend_address_pool", flattenAppGatewayBackendAddressPools(appGateway.ApplicationGatewayPropertiesFormat.BackendAddressPools))
+	d.Set("backend_http_settings", flattenAppGatewayBackendHTTPSettings(appGateway.ApplicationGatewayPropertiesFormat.BackendHTTPSettingsCollection))
+	d.Set("http_listener", flattenAppGatewayHTTPListeners(appGateway.ApplicationGatewayPropertiesFormat.HTTPListeners))
+	d.Set("probe", flattenAppGatewayProbes(appGateway.ApplicationGatewayPropertiesFormat.Probes))
+	d.Set("request_routing_rule", flattenAppGatewayRequestRoutingRules(appGateway.ApplicationGatewayPropertiesFormat.RequestRoutingRules))
+	d.Set("url_path_map", flattenAppGatewayURLPathMaps(appGateway.ApplicationGatewayPropertiesFormat.URLPathMaps))
+	d.Set("authentication_certificate", schema.NewSet(hashAppGatewayAuthenticationCertificates, flattenAppGatewayAuthenticationCertificates(appGateway.ApplicationGatewayPropertiesFormat.AuthenticationCertificates)))
+	d.Set("ssl_certificate", schema.NewSet(hashAppGatewaySslCertificates, flattenAppGatewaySslCertificates(appGateway.ApplicationGatewayPropertiesFormat.SslCertificates)))
+
+	if appGateway.ApplicationGatewayPropertiesFormat.WebApplicationFirewallConfiguration != nil {
+		d.Set("waf_configuration", schema.NewSet(hashAppGatewayWafConfig,
+			flattenAppGatewayWafConfig(appGateway.ApplicationGatewayPropertiesFormat.WebApplicationFirewallConfiguration)))
+	}
+
+	flattenAndSetTags(d, appGateway.Tags)
+
+	return nil
+}
+
+func resourceArmAppGatewayDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).appGatewayClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return errwrap.Wrapf("Error Parsing Azure Resource ID {{err}}", err)
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["applicationGateways"]
+
+	_, err = client.Delete(resGroup, name, make(chan struct{}))
+	if err != nil {
+		return errwrap.Wrapf("Error Deleting AppGateway {{err}}", err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func appGatewayResGroupAndNameFromID(appGatewayID string) (string, string, error) {
+	id, err := parseAzureResourceID(appGatewayID)
+	if err != nil {
+		return "", "", err
+	}
+	name := id.Path["applicationGateways"]
+	resGroup := id.ResourceGroup
+
+	return resGroup, name, nil
+}
+
+func retrieveAppGatewayById(appGatewayID string, meta interface{}) (*network.ApplicationGateway, bool, error) {
```

Comment:
do we need the exists argument here? I believe we should be able to infer it from the `ApplicationGateway` being nil?